src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1693,6 +1693,8 @@ private bool OpenJsonFileFromPath(string fileContents, string filePath, bool for
                             OnComputeModelDeserialized();
 
                             SetPeriodicEvaluation(ws);
+
+                            CurrentWorkspace = ws;
```

Comment:
Update the current workspace here, after the VM is re-initialized and all libs/packages are reloaded. This will trigger the property changed event for `CurrentWorkspace` at the right time for the `DynamoPackages` extension to react to it and load packages for that new workspace.